src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.RegularCompilationTracker_Generators.cs

```diff
@@ -36,49 +36,71 @@ private sealed partial class RegularCompilationTracker : ICompilationTracker
             Compilation? compilationWithStaleGeneratedTrees,
             CancellationToken cancellationToken)
         {
-            if (creationPolicy.GeneratedDocumentCreationPolicy is GeneratedDocumentCreationPolicy.DoNotCreate)
+            // First try to compute the SG docs in the remote process (if we're the host process), syncing the results
+            // back over to us to ensure that both processes are in total agreement about the SG docs and their
+            // contents.
+            var result = await TryComputeNewGeneratorInfoInRemoteProcessAsync(
```

Comment:
Should we be doing a check for "required" generators before we do this call in the first place?

Reply:
Yeah, I put a TODO in for now, so we can think about it. I'm not sure if we have the ability to actually tell what generators we have without going to OOP? Like we might have to go to OOP to find out if we need to go to OOP? I suppose we could cache that and only update if the analyzer refs change. Not sure if it's worth it. 

I guess let's do an RPS/Speedometer run and see if the numbers are noticeable and let that guide us?